icu4j/main/classes/core/src/com/ibm/icu/impl/ICUResourceBundle.java

```diff
@@ -1340,7 +1392,145 @@ public ICUResourceBundle load() {
+            if (b == null) {
+                b = normalLookup;
+            } else {
+                overrideResc = b;
+                if (normalLookup != null) {
+                    b.setParent(normalLookup);
+                    b = normalLookup;
+                }
+            }
+
+            // fallback to locale ID parent
+            if(b == null){
+                int i = localeName.lastIndexOf('_');
+                if (i != -1) {
+                    // Chop off the last underscore and the subtag after that.
+                    String temp = localeName.substring(0, i);
+                    b = instantiateBundle(baseName, temp, defaultID, root, openType);
+                }else{
+                    // No underscore, only a base language subtag.
+                    if(openType == OpenType.LOCALE_DEFAULT_ROOT &&
+                            !localeIDStartsWithLangSubtag(defaultID, localeName)) {
+                        // Go to the default locale before root.
+                        b = instantiateBundle(baseName, defaultID, defaultID, root, openType);
+                    } else if(openType != OpenType.LOCALE_ONLY && !rootLocale.isEmpty()) {
+                        // Ultimately go to root.
+                        b = ICUResourceBundle.createBundle(baseName, rootLocale, root, ICUBinary.DataFilesCategory.OVERRIDE);
+                    }
+                }
+            }else{
+                UResourceBundle parent = null;
+                localeName = b.getLocaleID();
+                int i = localeName.lastIndexOf('_');
+
+                // TODO: C++ uresbund.cpp also checks for %%ParentIsRoot. Why not Java?
+                String parentLocaleName = ((ICUResourceBundleImpl.ResourceTable)b).findString("%%Parent");
+                if (parentLocaleName != null) {
+                    parent = instantiateBundle(baseName, parentLocaleName, defaultID, root, openType);
+                } else if (i != -1) {
+                    parent = instantiateBundle(baseName, localeName.substring(0, i), defaultID, root, openType);
+                } else if (!localeName.equals(rootLocale)){
+                    parent = instantiateBundle(baseName, rootLocale, defaultID, root, openType);
+                }
+
+                if (!b.equals(parent)){
+                    b.setParent(parent);
+                }
+            }
+
+            if (overrideResc != null) {
+                return overrideResc;
+            } else {
+                return b;
+            }
+        }
+    }
+
+    /* Loads the ResourceBundle for the localeID and then does so recursively via the Loader for
+     * all locales in the parent chain from the given locale to the root locale.
+     */
+    private static ICUResourceBundle instantiateBundle(
+            final String baseName, final String localeID, final String defaultID,
+            final ClassLoader root, final OpenType openType) {
```

Comment:
We should probably check the openType, and look for overrides either only for the standard type, or else for all but the "direct" types.

Reply:
Done. Is this change how you were describing it?